source/common/network/io_socket_handle_impl.cc

```diff
@@ -4,56 +4,86 @@
 
 #include <iostream>
 
-#include "common/common/assert.h"
+#include "envoy/buffer/buffer.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/common/stack_array.h"
+#include "common/network/io_socket_error_impl.h"
 
 using Envoy::Api::SysCallIntResult;
 using Envoy::Api::SysCallSizeResult;
 
 namespace Envoy {
 namespace Network {
 
-Api::IoError::IoErrorCode IoSocketError::errorCode() const {
-  switch (errno_) {
-  case EAGAIN:
-    // EAGAIN should use specific error ENVOY_ERROR_AGAIN.
-    NOT_REACHED_GCOVR_EXCL_LINE;
-  case ENOTSUP:
-    return IoErrorCode::NoSupport;
-  case EAFNOSUPPORT:
-    return IoErrorCode::AddressFamilyNoSupport;
-  case EINPROGRESS:
-    return IoErrorCode::InProgress;
-  case EPERM:
-    return IoErrorCode::Permission;
-  default:
-    return IoErrorCode::UnknownError;
-  }
-}
-
-std::string IoSocketError::errorDetails() const { return ::strerror(errno_); }
-
 IoSocketHandleImpl::~IoSocketHandleImpl() {
   if (fd_ != -1) {
     IoSocketHandleImpl::close();
   }
 }
 
-// Deallocate memory only if the error is not ENVOY_ERROR_AGAIN.
-void deleteIoError(Api::IoError* err) {
-  ASSERT(err != nullptr);
-  if (err != ENVOY_ERROR_AGAIN) {
-    delete err;
-  }
-}
-
 Api::IoCallUintResult IoSocketHandleImpl::close() {
   ASSERT(fd_ != -1);
   const int rc = ::close(fd_);
   fd_ = -1;
-  return Api::IoCallResult<uint64_t>(rc, Api::IoErrorPtr(nullptr, deleteIoError));
+  return Api::IoCallResult<uint64_t>(rc, Api::IoErrorPtr(nullptr, IoSocketError::deleteIoError));
```

Comment:
IOCallUint64Result

Reply:
and my main point here is that you defined a nickname but didn't use it here.

Reply:
I vaguely remember constructing a unique_ptr using alias `SomeTypePtr` is not preferred. But since you requested, done.